storage/storage.go

```diff
@@ -1341,6 +1349,11 @@ type Query struct {
 	// lexicographically before endOffset. If startOffset is also set, the objects
 	// listed will have names between startOffset (inclusive) and endOffset (exclusive).
 	EndOffset string
+
+	// Projection defines the set of properties to return. It will default to ProjectionFull,
+	// which returns all properties. Passing ProjectionNoACL will omit Owner and ACL,
+	// which may improve performance when listing many objects.
+	Projection string
```

Comment:
Could this ever have more then two possible values @tritone ? Maybe this should be a named iota?

Reply:
Discussed with Cody, this seems like a good idea. Let's do the following:

```
type Projection int

const (
  ProjectionDefault  Projection = iota
  ProjectionFull
  ProjectionNoACL
)

func (p Projection) string() {
  // return "", "full", "noAcl" respectively
}
```

Reply:
Why do we need a separate `ProjectionDefault`? We can simply use `ProjectionFull` as default.

Reply:
@codyoss suggested this to follow https://google.aip.dev/126 . I do think it seems good to disambiguate between the user not specifying a projection and the user specifically asking for "full" in case API values or defaults change in the future.

Reply:
I see.
Do we really need to return an empty string for default projection? I guess we need a string only when we pass projection to generated `ObjectsListCall`, so there should be only "full" and "noAcl".

Reply:
Yes, I think it's best to have it for completeness. @codyoss feel free to chime in if you have thoughts.